Solutions/Marain.Operations.Api.Specs/Steps/OperationsUsingClientSteps.cs

```diff
@@ -0,0 +1,262 @@
+﻿// <copyright file="OperationsUsingClientSteps.cs" company="Endjin Limited">
+// Copyright (c) Endjin Limited. All rights reserved.
+// </copyright>
+
+namespace Marain.Operations.Api.Specs.Steps
+{
+    using System;
+    using System.Net;
+    using System.Threading.Tasks;
+    using Corvus.Testing.SpecFlow;
+    using Marain.Operations.Api.Specs.Bindings;
+    using Marain.Operations.Client.OperationsControl;
+    using Marain.Operations.Client.OperationsControl.Models;
+    using Marain.Operations.Client.OperationsStatus;
+    using Marain.Operations.Client.OperationsStatus.Models;
+    using Marain.TenantManagement.Testing;
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Rest;
+    using NUnit.Framework;
+    using TechTalk.SpecFlow;
+
+    [Binding]
+    public class OperationsUsingClientSteps : Steps
+    {
+        public const string LastResponseContextKey = "LastResponse";
+
+        [When("I use the operations status client to get the operation with the Id called '(.*)'")]
+        public Task WhenIUseTheOperationsStatusClientToGetTheOperationWithTheIdCalled(string operationIdName)
+        {
+            Guid operationId = this.ScenarioContext.Get<Guid>(operationIdName);
+
+            return this.WhenIUseTheOperationsStatusClientToGetTheOperationWithId(operationId);
```

Comment:
The advice posted by (I think) Martin Fowler recently about `Task` usage said don't do this. Turn this into an `async` method and replace the `return` with an `await` here. The main problem in this case is that if an exception occurs, as it stands you'll lose the async pseudo stack trace - it'll look like it threw inside SpecFlow.

Same for a few other methods in the class.

Reply:
Yes, we've been doing it this way for a while, and have never habituated doing it the better way.